source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -123,8 +128,35 @@ void ZooKeeperFilterConfig::initOpCode(OpCodes opcode, Stats::Counter& resp_coun
 }
 
 int32_t ZooKeeperFilterConfig::getOpCodeIndex(LatencyThresholdOverride_Opcode opcode) {
-  auto it = opcodeMap().find(opcode);
-  if (it != opcodeMap().end()) {
+  static OpcodeMap opcode_map = {{LatencyThresholdOverride::Connect, 0},
```

Comment:
this pattern causes a non-deterministic destructor order fiasco. The previous impl you had was closer to what I was thinking. Just need a temp for the map:
```
const OpcodeMap& opcode_map = opcodeMap();
auto it = opcode_map.find(opcode);
if (it != opcode_map.end()) {
...
```

Reply:
Sorry, I misunderstood your previous comment, just updated the code.